core/src/data_sources/data_source.rs

```diff
@@ -438,6 +452,8 @@ impl DataSource {
             .map(|tag| tag.to_string())
             .collect();
 
+        let parents: Vec<String> = parents.iter().map(|parent| parent.to_string()).collect();
```

Comment:
I believe you don't need that line. The one above is due to the merging of the arrays but you should be good to remove this one no?